core-api/src/main/java/com/optimizely/ab/Optimizely.java

```diff
@@ -711,8 +712,19 @@ public Variation getVariation(@Nonnull Experiment experiment,
                                   @Nonnull String userId,
                                   @Nonnull Map<String, ?> attributes) throws UnknownExperimentException {
         Map<String, ?> copiedAttributes = copyAttributes(attributes);
+        Variation variation = decisionService.getVariation(experiment, userId, copiedAttributes);
+
+        Map<String, Object> decisionInfo = new HashMap<>();
+        decisionInfo.put(DecisionInfoEnums.ExperimentDecisionInfo.EXPERIMENT_KEY.toString(), experiment.getKey());
```

Comment:
Similar comment in https://github.com/optimizely/java-sdk/pull/270 on how to make this cleaner.